tests/app/views/test_marketplace.py

```diff
@@ -477,25 +479,123 @@ def setup(self):
+
+        ss_elem = document.xpath("//p[@class='search-summary']")[0]
+        assert self._normalize_whitespace(self._squashed_element_text(ss_elem)) == "2 opportunities"
+
+    def test_catalogue_of_briefs_page_filtered(self):
+        original_url = "/digital-outcomes-and-specialists/opportunities?page=2&status=live&lot=lot-one&lot=lot-three"
+        res = self.client.get(original_url)
+        assert_equal(200, res.status_code)
+        document = html.fromstring(res.get_data(as_text=True))
+
+        self._data_api_client.get_framework.assert_called_once_with("digital-outcomes-and-specialists")
+        self._data_api_client.find_briefs.assert_called_once()
+        regular_args = {
+            k: v for k, v in iteritems(self._data_api_client.find_briefs.call_args[1]) if k not in ("status", "lot",)
+        }
+        assert regular_args == {
+            "framework": "digital-outcomes-and-specialists",
+            "page": 2,
+            "human": True,
+        }
+        assert set(self._data_api_client.find_briefs.call_args[1]["status"].split(",")) == {"live"}
+        assert set(self._data_api_client.find_briefs.call_args[1]["lot"].split(",")) == {"lot-one", "lot-three"}
+
+        heading = document.xpath('normalize-space(//h1/text())')
+        assert heading == "Digital Outcomes and Specialists opportunities"
+        assert 'lot 1, lot 3 and lot 4' in document.xpath(
+            "normalize-space(//div[@class='marketplace-paragraph']/p/text())"
+        )
+
+        lot_inputs = document.xpath("//form[@method='get']//input[@name='lot']")
+        assert {
+            element.get("value"): bool(element.get("checked"))
+            for element in lot_inputs
+        } == {
+            "lot-one": True,
+            "lot-three": True,
+            "lot-four": False,
+        }
+
+        status_inputs = document.xpath("//form[@method='get']//input[@name='status']")
+        assert {
+            element.get("value"): bool(element.get("checked"))
+            for element in status_inputs
+        } == {
+            "live": True,
+            "closed": False,
+        }
+
+        parsed_original_url = urlparse(original_url)
+        parsed_prev_url = urlparse(document.xpath("//li[@class='previous']/a/@href")[0])
+        parsed_next_url = urlparse(document.xpath("//li[@class='next']/a/@href")[0])
+        assert parsed_original_url.path == parsed_prev_url.path == parsed_next_url.path
+
+        normalize_qs = lambda qs: {k: set(v) for k, v in iteritems(parse_qs(qs)) if k != "page"}
+        assert normalize_qs(parsed_original_url.query) == \
+            normalize_qs(parsed_next_url.query) == \
+            normalize_qs(parsed_prev_url.query)
+
+        ss_elem = document.xpath("//p[@class='search-summary']")[0]
+        assert self._normalize_whitespace(self._squashed_element_text(ss_elem)) == "2 results"
+
+    def test_catalogue_of_briefs_400_if_invalid_status(self):
```

Comment:
This should be a 404 rather than 400 - there's nothing wrong with the HTTP request itself - we can understand what's being asked for - it's just that we don't currently have anything with that status.